derive_builder/src/lib.rs

```diff
@@ -367,6 +367,58 @@
 //! }
 //! ```
 //!
+//! ## Build Method Customization
+//! You can rename or suppress the auto-generated build method, leaving you free to implement 
+//! your own version. Suppression is done using `#[builder(build_fn(skip))]` at the struct level,
+//! and renaming is done with `#[builder(build_fn(name = "YOUR_NAME"))]`.
+//!
+//! ## Pre-Build Validation
+//! If you're using the provided `build` method, you can declare 
+//! `#[builder(build_fn(validator="path::to::fn"))]` to specify a validator function which gets
+//! access to the builder before construction. 
+//! 
+//! The provided function must have the signature `(&FooBuilder) -> Result<_, String>`; 
+//! the `Ok` variant is not used by the `build` method, and must be accessible from the scope
+//! where the target struct is declared. The path does not need to be fully-qualified, and will
```

Comment:
Please also add a note somewhere, that default values are _not_ validated, i.e. the validation is run before default values are applied. This may not be obvious.